gems/bess/lib/commands/houdini/nonprofit/create/create_command.rb

```diff
@@ -1,38 +1,62 @@
+			option :confirm_admin, default: true, type: :boolean, desc: "Require the nonprofit admin to be confirmed via email"
+
+			option :nonprofit_name, default: nil, desc: "Provide the nonprofit's name"
+			option :state_code, default: nil, desc: "Provide the nonprofit' state code"
+			option :city, default: nil, desc: "Provide the nonprofit's city"
+			option :nonprofit_website, default: nil, desc: "Provide the nonprofit public website"
+			option :nonprofit_email, default: nil, desc: "Provide the nonprofit public email"
+
+			option :user_name, default: nil, desc: "Provide the nonprofit's admin's name"
+			option :user_email, default: nil, desc: "Provide the nonprofit's admin's email address (It'll be used for logging in)"
+			option :user_phone, default: nil, desc: "[OPTIONAL] Provide the nonprofit's admin's phone"
+			option :user_password, default: nil, desc: "Provide the nonprofit's admin's password"
+
 			def perform
-				
 				result = {
 					nonprofit: {
-						name: ask("What is the nonprofit's name?"),
-						state_code: ask("What is the nonprofit's state?"),
-						city: ask("What is the nonprofit's city?"),
+						name: options[:nonprofit_name] || ask("What is the nonprofit's name?"),
+						state_code: options[:state_code] || ask("What is the nonprofit's state?"),
+						city: options[:city] || ask("What is the nonprofit's city?"),
+						website: options[:nonprofit_website] || ask("[OPTIONAL] What is the nonprofit's public website?"),
+						email: options[:nonprofit_email] || ask("[OPTIONAL] What is the nonprofit's public e-mail?")
 					}, 
 					user: {
-						name: ask("What is your nonprofit's admin's name?"),
-						email: ask("What is your nonprofit's admin's email address (it'll be used for logging in)?"),
-						password: ask("What is the nonprofit's admin's password?", echo: false)
+						name: options[:user_name] || ask("What is your nonprofit's admin's name?"),
+						email: options[:user_email] || ask("What is your nonprofit's admin's email address? (It'll be used for logging in)"),
+						phone: options[:user_phone] || ask("[OPTIONAL] What is your nonprofit's admin's phone number?"),
+						password: options[:user_password] || ask("What is the nonprofit's admin's password?", echo: false)
 					}
 				}
 				say
 				require_application_and_environment!
-				nonprofit = ::Nonprofit.new(result[:nonprofit].merge({register_np_only: true}))
-				user = nonprofit.admins.build(result[:user])
-				unless nonprofit.save
-					nonprofit.errors.full_messages.each do |i|
-						say i
+
+				result = sanitize_optional_fields(result)
+
+				nonprofit = ::Nonprofit.new(result[:nonprofit].merge({ register_np_only: true }))
+				user = User.new(result[:user])
+				user.skip_confirmation! if options[:dont_confirm_admin]
+				roles = [Role.new(host: nonprofit, name: 'nonprofit_admin', user: user)]
+				roles << Role.new(host: nonprofit, name: 'super_admin', user: user) if options[:super_admin]
```

Comment:
Did you ever try to use `build_admin` again after we corrected the roles names? If it it work, it would allow us to simplify this code a bit.


It might make writing out the errors below a bit more difficult though?

Reply:
@clarissalimab Anymore thought given to this?

Reply:
It didn't work with `nonprofit.admins.build`, with or without the `super-admin` argument - it said that Roles is invalid.

Reply:
Hmmm, fair enough. We'll just go with this then!